src/itemstackmetadata.cpp

```diff
@@ -58,15 +59,28 @@ bool ItemStackMetadata::setString(const std::string &name, const std::string &va
 	return result;
 }
 
-void ItemStackMetadata::serialize(std::ostream &os) const
+void ItemStackMetadata::serialize(std::ostream &os, bool disk) const
```

Comment:
Not obvious what this does. I'd recommend inverting the meaning: `sparse_serialization` or `shallow_serialization`